Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilderUtils.cpp

```diff
@@ -24,12 +24,9 @@ namespace AZ::RPI::MaterialBuilderUtils
         AZStd::optional<AZ::u32> productSubId)
     {
         bool dependencyFileFound = false;
-            
-        const bool currentFileIsMaterial = AzFramework::StringFunc::Path::IsExtension(currentFilePath.c_str(), MaterialSourceData::Extension);
-        const bool referencedFileIsMaterialType = AzFramework::StringFunc::Path::IsExtension(referencedParentPath.c_str(), MaterialTypeSourceData::Extension);
-        const bool shouldFinalizeMaterialAssets = MaterialBuilderUtils::BuildersShouldFinalizeMaterialAssets();
 
-        AZStd::vector<AZStd::string> possibleDependencies = RPI::AssetUtils::GetPossibleDepenencyPaths(currentFilePath, referencedParentPath);
+        AZStd::vector<AZStd::string> possibleDependencies =
+            RPI::AssetUtils::GetPossibleDepenencyPaths(currentFilePath, referencedParentPath);
```

Comment:
performance nit: function returns a whole vector for 2 strings. Would be great to update to a fixed_vector of fixed_string's so no allocations are needed.